src/modules/cron/cron.controller.ts

```diff
@@ -73,36 +74,35 @@ export class CronController {
     @ApiInternalServerErrorResponse({
         description: 'Internal server error while trying to remove personenkontexte from users.',
     })
-    public async removePersonenKontexteWithExceedingLimitFromUsers(
+    public async removePersonenKontexteWithExpiredBefristungFromUsers(
         @Permissions() permissions: PersonPermissions,
     ): Promise<boolean> {
         try {
             //Get PersonenKontexte for a person at least one kontext exceeds the befristung value
-            const personenKontexteGroupedByPersonId: Record<string, Personenkontext<true>[]> =
-                await this.personenKonextRepository.getPersonenKontexteWithExceedingBefristung();
-            if (Object.keys(personenKontexteGroupedByPersonId).length === 0) {
+            const personenKontexteGroupedByPersonId: Map<PersonID, Personenkontext<true>[]> =
+                await this.personenKonextRepository.getPersonenKontexteWithExpiredBefristung();
+            if (personenKontexteGroupedByPersonId.size === 0) {
                 return true;
             }
 
-            //filter each Personenkontext and validate Personenkontexte to keep for each person
+            // Filter each Personenkontext and validate Personenkontexte to keep for each person
             const promises: Promise<Personenkontext<true>[] | PersonenkontexteUpdateError>[] = [];
-            for (const personId in personenKontexteGroupedByPersonId) {
-                const personenKontexte: Personenkontext<true>[] | undefined =
-                    personenKontexteGroupedByPersonId[personId];
-
-                if (Array.isArray(personenKontexte)) {
-                    const count: number = personenKontexte.length;
-                    //filter PersonenKontexte
-                    const personenKontexteToKeep: DbiamPersonenkontextBodyParams[] =
-                        this.filterPersonenKontexte(personenKontexte);
-                    //validate PersonenKontexte to keep
-                    promises.push(
-                        this.personenkontextWorkflowFactory
-                            .createNew()
-                            .commit(personId, new Date(), count, personenKontexteToKeep, permissions),
-                    );
-                }
-            }
+            personenKontexteGroupedByPersonId.forEach(
+                (personenKontexte: Personenkontext<true>[], personId: PersonID) => {
+                    if (personenKontexte) {
```

Comment:
`personenKontexte` should always be an array, so this check will always be `true` and serves no purpose.

Reply:
get a "might be undefined error"

Reply:
Weird, in the line above the type says it should only ever be `Personenkontext<true>[]`